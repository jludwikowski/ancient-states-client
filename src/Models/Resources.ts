import Resource from './Resource';

export default interface Resources {
    [key:string]: any;
    food: Resource;
    iron: Resource;
    timber: Resource;
    stone: Resource;
    gold: Resource;
    influence: Resource;
}
