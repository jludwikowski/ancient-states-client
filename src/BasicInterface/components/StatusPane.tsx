import React from 'react';
import Resources from '../../Models/Resources';

export const StatusPane: React.FunctionComponent<Resources> = (props) => (
     <div className='sidebar-sticky'>
         <ul className='flex-column'>
             { Object.keys(props).map((key:any, index) => {
                     return (
                         <li className='item' key={index}>
                             <span>{props[key].label}</span><span>: {props[key].value}</span>
                         </li>
                     )
                 }
             )}
        </ul>
    </div>

)
