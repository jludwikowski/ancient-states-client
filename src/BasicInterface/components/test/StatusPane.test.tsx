import { shallow } from 'enzyme';
import React from 'react';
import ReactDOM from 'react-dom';
import Resources from "../../../Models/Resources";
import { StatusPane } from '../StatusPane';

describe('StatusPane', () => {

    const resources:Resources = {
        id: 1,
        food: 10000,
        iron: 1000,
        timber: 1000,
        stone: 1000,
        gold: 10000,
        influence: 100,
        owner: 1,
    };

    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<StatusPane {...resources} />, div);
        ReactDOM.unmountComponentAtNode(div);
    });

    it('Should match snapshot', () => {
        const component = shallow(<StatusPane {...resources} />);
        expect(component).toMatchSnapshot();
    });
});
