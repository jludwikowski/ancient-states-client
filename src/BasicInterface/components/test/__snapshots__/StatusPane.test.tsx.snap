// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StatusPane Should match snapshot 1`] = `
<div
  className="sidebar-sticky"
>
  <ul
    className="flex-column nav"
  >
    <li
      className="item"
      key="0"
    >
      <img
        className="icon"
        src="images/icons/icons8-wheat-30.png"
      />
      <span>
        Food
      </span>
      <span>
        : 
        10000
      </span>
    </li>
    <li
      className="item"
      key="1"
    >
      <img
        className="icon"
        src="images/icons/icons8-steel-bars-48.png"
      />
      <span>
        Iron
      </span>
      <span>
        : 
        1000
      </span>
    </li>
    <li
      className="item"
      key="2"
    >
      <img
        className="icon"
        src="images/icons/icons8-wood-64.png"
      />
      <span>
        Timber
      </span>
      <span>
        : 
        1000
      </span>
    </li>
    <li
      className="item"
      key="3"
    >
      <img
        className="icon"
        src="images/icons/icons8-rock-30.png"
      />
      <span>
        Stone
      </span>
      <span>
        : 
        1000
      </span>
    </li>
    <li
      className="item"
      key="4"
    >
      <img
        className="icon"
        src="images/icons/icons8-coins-30.png"
      />
      <span>
        Gold
      </span>
      <span>
        : 
        10000
      </span>
    </li>
    <li
      className="item"
      key="5"
    >
      <img
        className="icon"
        src="images/icons/icons8-museum-30.png"
      />
      <span>
        Influence
      </span>
      <span>
        : 
        100
      </span>
    </li>
  </ul>
</div>
`;
